website/docs/externaldata.md

```diff
@@ -306,3 +307,99 @@ There are several limitations when using external data with the mutating webhook
 - `ModifySet` does not support external data.
 - Multiple mutations to the same object are applied alphabetically based on the name of the mutation CRDs. If you have an external data mutation and a non-external data mutation with the same `spec.location`, the final result might not be what you expected. Currently, there is no way to enforce custom ordering of mutations but the issue is being tracked [here](https://github.com/open-policy-agent/gatekeeper/issues/1133).
+
+## TLS and mutual TLS support
+
+Since external data providers are in-cluster HTTP servers backed by Kubernetes services, communication is not encrypted by default. This can potentially lead to security issues such as request eavesdropping, tampering, and man-in-the-middle attack.
+
+To further harden the security posture of the external data feature, starting from Gatekeeper v3.9.0, TLS and mutual TLS (mTLS) via HTTPS protocol are supported between Gatekeeper and external data providers. In this section, we will describe the steps required to configure them.
+
+### Prerequisites
+
+- A Gatekeeper deployment **v3.9.0+**.
+- The certificate of your certificate authority (CA) in PEM format.
+- The certificate of your external data provider in PEM format, signed by the CA above.
+- The private key of the external data provider in PEM format.
+
+### (Optional) How to generate a self-signed CA and a keypair for the external data provider
+
+In this section, we will describe how to generate a self-signed CA and a keypair using `openssl`.
+
+1. Generate a private key for your CA:
+
+```bash
+openssl genrsa -out ca.key 2048
+```
+
+2. Generate a self-signed certificate for your CA:
+
+```bash
+openssl req -new -x509 -days 365 -key ca.key -subj "/O=My Org/CN=External Data Provider CA" -out ca.crt
+```
+
+3. Generate a private key for your external data provider:
+
+```bash
+ openssl genrsa -out server.key 2048
+```
+
+4. Generate a certificate signing request (CSR) for your external data provider:
+
+> Replace `<service name>` and `<service namespace>` with the correct values.
+
+```bash
+openssl req -newkey rsa:2048 -nodes -keyout server.key -subj "/CN=<service name>.<service namespace>" -out server.csr
+```
+
+5. Generate a certificate for your external data provider:
+
+```bash
+openssl x509 -req -extfile <(printf "subjectAltName=DNS:<service name>.<service namespace>") -days 365 -in server.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out server.crt
+```
+
+### How Gatekeeper trusts the external data provider (TLS)
+
+To enable one-way TLS, your external data provider should enable any TLS-related configurations for their HTTP server. For example, for Go's built-in [`HTTP server`](https://pkg.go.dev/net/http#Server) implementation, you can use [`ListenAndServeTLS`](https://pkg.go.dev/net/http#ListenAndServeTLS).
+
```

Comment:
can we add an example here? like https://github.com/open-policy-agent/gatekeeper/blob/ff9f2cd297310b0cb9956ec5994ae148f8d93e57/test/externaldata/dummy-provider/provider.go#L47